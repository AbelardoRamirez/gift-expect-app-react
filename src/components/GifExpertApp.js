import React, { useState } from 'react'
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="row justify-content-center">
            <div className="col">
                <h2 className="h2 font-weight-lighter text-center text-muted my-3">
                    GiftExpertApp
                </h2>
                <AddCategory setCategories={setCategories} />
                <hr className="hr shadow-lg" />
                <div className="row justify-content-center">
                    <div className="col">
                        {
                            categories.map(cat => (<GifGrid category={cat} key={cat} />))
                        }
                    </div>
                </div>
            </div>
        </div>
    )
}
export default GifExpertApp