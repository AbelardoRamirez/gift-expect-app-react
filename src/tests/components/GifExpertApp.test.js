import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../../components/GifExpertApp';

describe('Pruebas en el componente <GifExpertApp/>', () => {
    test('Debe de renderizar correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Programming', 'VueJs'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });


});