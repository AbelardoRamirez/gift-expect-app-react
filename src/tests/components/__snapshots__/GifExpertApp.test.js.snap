// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col"
  >
    <h2
      className="h2 font-weight-lighter text-center text-muted my-3"
    >
      GiftExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr
      className="hr shadow-lg"
    />
    <div
      className="row justify-content-center"
    >
      <div
        className="col"
      >
        <GifGrid
          category="Programming"
          key="Programming"
        />
        <GifGrid
          category="VueJs"
          key="VueJs"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en el componente <GifExpertApp/> Debe de renderizar correctamente el componente 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col"
  >
    <h2
      className="h2 font-weight-lighter text-center text-muted my-3"
    >
      GiftExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr
      className="hr shadow-lg"
    />
    <div
      className="row justify-content-center"
    >
      <div
        className="col"
      />
    </div>
  </div>
</div>
`;
